ec/ecresource/deploymentresource/elasticsearch/v2/node_roles_plan_modifier.go

```diff
@@ -20,17 +20,19 @@ package v2
 import (
 	"context"
 
-	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 )
 
-func UseNodeRolesDefault() tfsdk.AttributePlanModifier {
+func UseNodeRolesDefault() nodeRolesDefault {
```

Comment:
The previous return value was also effectively a type assertion. 

Might be worth adding something like this to the top of this file:

```
var _ planmodifier.Set = nodeRolesDefault{}
```